lib/mongo/uri.rb

```diff
@@ -615,6 +626,94 @@ def auth_mech_props(value)
       properties
     end
 
+    # Parses the zlib compression level.
+    #
+    # @param value [ String ] The zlib compression level string.
+    #
+    # @return [ Integer | nil ] The compression level value if it is between -1 and 9 (inclusive),
+    #   otherwise nil (and a warning will be raised).
+    def zlib_compression_level(value)
+      if /^-?\d+$/ =~ value
+        i = value.to_i
+
+        if i >= -1 && i <= 9
+          return i
+        end
+      end
+
+      log_warn("#{value} is not a valid zlibCompressionLevel")
+      nil
+    end
+
+
+    # Parses a boolean value.
+    #
+    # @param value [ String ] The URI option value.
+    #
+    # @return [ true | false | nil ] The boolean value parsed out, otherwise nil (and a warning
+    #   will be raised).
+    def bool(value)
+      case value
+      when "true"
+        true
+      when "false"
+        false
+      else
+        log_warn("invalid boolean: #{value}")
+        nil
+      end
+    end
+
+    # Parses a boolean value and returns its inverse. This is used for options where the spec
+    # defines the boolean value semantics of enabling/disabling something in the reverse way that
+    # the driver does. For instance, the client option `ssl_verify` will disable certificate
+    # checking when the value is false, but the spec defines the option
+    # `tlsAllowInvalidCertificates`, which disables certificate checking when the value is true.
+    #
+    # @param value [ String ] The URI option.
+    #
+    # @return [ true | false | nil ] The inverse of boolean value parsed out, otherwise nil (and a
```

Comment:
Which means also options like tlsallowinvalidcertificates will always be parsed into true/false and not nil when not provided.

Reply:
Fixed